pgx-macros/src/lib.rs

```diff
@@ -628,18 +628,18 @@ fn impl_postgres_enum(ast: DeriveInput) -> proc_macro2::TokenStream {
         let label_string = label_ident.to_string();
 
         from_datum.extend(quote! { #label_string => Some(#enum_ident::#label_ident), });
-        into_datum.extend(quote! { #enum_ident::#label_ident => Some(::pgx::lookup_enum_by_label(#enum_name, #label_string)), });
+        into_datum.extend(quote! { #enum_ident::#label_ident => Some(__pgx__codegen__private__::lookup_enum_by_label(#enum_name, #label_string)), });
     }
 
     stream.extend(quote! {
-        impl ::pgx::datum::FromDatum for #enum_ident {
+        impl __pgx__codegen__private__::FromDatum for #enum_ident {
             #[inline]
-            unsafe fn from_polymorphic_datum(datum: ::pgx::pg_sys::Datum, is_null: bool, typeoid: ::pgx::pg_sys::Oid) -> Option<#enum_ident> {
+            unsafe fn from_polymorphic_datum(datum: __pgx__codegen__private__::Datum, is_null: bool, typeoid: __pgx__codegen__private__::Oid) -> Option<#enum_ident> {
```

Comment:
This makes reading the `cargo expand` versions of PGX code **much** worse and it doesn't feel like it actually improves anything. In addition, it creates another layer of indirection, instead of just stating the path that something came from, because of the reexports.

Reply:
we can bikeshed the name, but I'm convinced of real benefits to having all pgx symbols our code generation uses coming from a central place.

Reply:
> This makes reading the cargo expand versions of PGX code much worse and it doesn't feel like it actually improves anything

Possibly, but it's semi-important that we make it clear which things are intended to be user-accessible. `rustc`, `rust-analyzer`, and `clion`, will all happily tell you to use things in `#[doc(hidden)]` modules that are effectively private with no warning. This has caused big issues in the ecosystem before (a largeish crate (dashmap) was using private modules from serde due to this, and there was widespread breakage).

My feeling is that anybody using `cargo expand` on `pgx` is in for a night of horrors, but I hadn't really considered it. Certainly if you'd like it to be slightly less noisy while still being clear that it's private, I don't mind.